server/src/_utils/validate-subscriptions.ts

```diff
@@ -0,0 +1,24 @@
+const validCustomFieldNames = ["K01", "K02", "K03", "K04", "K05", "K06", "K07", "K08", "K09", "K10", "K11", "K12", "K13", "K14", "K15", "K16", "K17", "K18", "X01", "X02", "X03", "X04", "X05", "X06", "X07", "X08", "X09", "X10", "X11", "X12", "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10", "M11", "M12", "Q01", "Q02", "Q03", "Q04", "Q05", "Q06", "Q07", "Q08", "Q09", "Q10", "Q11", "Q12", "Q13", "Q14", "R01", "R02", "R03", "CW"];
```

Comment:
Two things here:

* I think I prefer something like `topics` for the name of this, just so we aren't bleeding Sendgrid terminology in. But I know that approach is subjective so I'll leave it up to you
* One easy change we _can_ make regardless of naming is to add some typing. You can create a [string literal type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) out of this by added `as const` at the end:

```
const topics = ["K01","K02",...] as const
``` 
Then create a type for strings that should be one of these string
```
type Topic = typeof topics[number];
```

Reply:
I don't think `topics` is a good name for this, because it doesn't really correspond to either the Sendgrid terminology nor ours.  I think `validCustomFieldNames` and `validCustomFieldValues` makes sense because any developer looking at this in the future would intuit that they are looking at a list of Community Board ids, and they correspond to custom field names in Sendgrid.

Reply:
> * One easy change we _can_ make regardless of naming is to add some typing. You can create a [string literal type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) out of this by added `as const` at the end:

I've made updates that I believe match what you're requesting, but I don't really understand the benefit of doing so.  You established that this is very specific to subscriptions, and so we aren't going to be exporting and using these types anywhere else.  Is there a benefit to adding all this code that gets compiled into the exact JS we had before?

Reply:
Sure, happy to explain. One benefit is simply developer ergonomics. If you have a variable or parameter set to this new type, and try to assign it to a string, you'll get auto-completion in VSCode to one of the possible values. It's a small thing but it's a nice benefit for not much work.

Another is how stricter typing gets you to think about possible error states you might otherwise not have accounted for. If you have a variable or parameter typed to this new type and you try to assign it to a regular old `string'. You might now know for sure that your string is one of the valid values, and typescript will yell at you for trying, which in turn might remind you to do validation, avoiding a possible bug. Again, its a small thing in this scenario but once you get used to thinking in types, it's not a ton of work and the benefits add up.